docs/language/functions/regexp.md

```diff
@@ -0,0 +1,34 @@
+### Function
+
+&emsp; **regexp** &mdash; perform a regular expression search on a string
+
+### Synopsis
+
+```
+regexp(re: string, s: string) -> any
+```
+### Description
+The _regexp_ function returns an array of strings holding the text
+of the left most match of the regular expression string re and the
+results of its sub expression if there are any. A null value indicates
+no match.
+
+### Examples
+
+Regexp returns an array of the match and its subexpressions:
+```mdtest-command
+echo '"seafood fool"' | zq -z 'yield regexp("foo(.?)", this) -'
```

Comment:
```suggestion
echo '"seafood fool friend"' | zq -z 'yield regexp("foo(.?) (\\w+) fr.*", this)' -
```

I think it'd be helpful for the example to show multiple capture groups to help users recognize that the approach is very similar to what they may have seen in Python, TCL, etc.